spec/gameSpec.js

```diff
@@ -0,0 +1,200 @@
+describe('game', function () {
+
+  var game;
+
+  beforeEach(function () {
+    game = new Game();
+  });
+
+  describe('Strike or Spare', function () {
+    it('Should increment frame counter by 1', function () {
+      game.pinsDroppedEachRoll(3);
+      expect(game.frameCounter).toEqual(1);
+    });
+
+    it('Should return true for a Strike', function () {
+      expect(game.isStrike(10)).toBeTruthy();
+    });
+
+    it('Score array should contain 10 and 0 for frame one - after strike', function () {
+      game.pinsDroppedEachRoll(10);
+
+      expect(game.scoreArray).toEqual([10,0]);
+    });
+
+    it('Should return true if a Spare', function () {
+      game.pinsDroppedEachRoll(4);
+
+      expect(game.isSpare(6)).toBeTruthy();
+    });
+
+    it('Neither spare or strike with 4 and 5 rolls', function () {
+      game.pinsDroppedEachRoll(4);
+
+      expect(game.isSpare(5)).toBeFalsy();
+    });
+
+    it('Score array should contain 4 and 5 rolls', function () {
+      game.pinsDroppedEachRoll(4);
+      game.pinsDroppedEachRoll(5);
+
+      expect(game.scoreArray).toEqual([4,5]);
+    });
+
+  });
+
+  describe('Total score after each frame', function () {
+    it('Return total score for 3 frames: 4 and 5 rolls', function () {
+      game.startGame(5);
+      game.startGame(4);
+
+      game.startGame(5);
+      game.startGame(4);
+
+      game.startGame(5);
+      game.startGame(4);
+
+      expect(game.finalScore).toEqual(27);
+    });
+
+    it('Return total score for 2 frames: strike followed by 4 and 5 rolls', function () {
+      game.startGame(10);
+      game.startGame(4);
+      game.startGame(5);
+
+      expect(game.finalScore).toEqual(28);
+    });
+
+    it('Return total score for 3 frames: 2 strikes followed by 4 and 4 rolls', function () {
+      game.startGame(10);
+      game.startGame(10);
+      game.startGame(4);
+      game.startGame(4);
+
+      expect(game.finalScore).toEqual(50);
+    });
+
+    it('Return total score for 3 strikes followed by: 3,4 and 5,2 rolls', function () {
+      game.startGame(10);
+      game.startGame(10);
+      game.startGame(10);
+      game.startGame(3);
+      game.startGame(4);
+      game.startGame(5);
+      game.startGame(2);
+      expect(game.finalScore).toEqual(84);
+    });
+
+    it('Return total score for spare (0,10) and a strike followed by: 3,4 and 5,2 rolls', function () {
+      game.startGame(0);
+      game.startGame(10);
+      game.startGame(10);
+      game.startGame(5);
+      game.startGame(2);
+      game.startGame(4);
+      game.startGame(1);
+
+      expect(game.finalScore).toEqual(49);
+    });
+
+    it('Return total score for 10 frames without a tenth frame bonus', function () {
+      game.startGame(0);
+      game.startGame(10);
+
+      game.startGame(10);
```

Comment:
Long tests like this are usually either an indication that (a) we're testing the wrong thing or (b) our classes are too tightly coupled. I think it's a bit of both here. 

Regards (a) - are we really testing anything here that we aren't effectively testing in other tests? Specifically, the test on line 46 looks like a shorter version of this. I think what we want to test is that we can calculate the total score - in which case, if it works for 2 or 3 frames, testing for more frames doesn't really add anything. 

Regards (b) - you only have one class, so suggesting that classes are tightly coupled may sound strange, but it feels like there is logic here that should be refactored out of Game. If I had a separate Frame class, for example, and if I was injecting that into Game, then I could write a test something like:

```
var frames = [new Frame(0, 10), new Frame(7, 2)];
var totalNumberOfFrames = 2;
var game = new Game(frames, totalNumberOfFrames);

var totalScore = game.getTotalScore();

expect(totalScore).toBe(19);
```

Here I am effectively testing the same case but it is terser and easier to read and also more flexible.